modules/store/src/selector.ts

```diff
@@ -17,13 +17,25 @@ export type ComparatorFn = (a: any, b: any) => boolean;
 
 export type DefaultProjectorFn<T> = (...args: any[]) => T;
 
+export type ProjectorFn<Args extends unknown[], T> = (...args: Args) => T;
+
 export interface MemoizedSelector<
   State,
   Result,
   ProjectorFn = DefaultProjectorFn<Result>
 > extends Selector<State, Result> {
   release(): void;
-  projector: ProjectorFn;
+  /** @deprecated A projector must take at least 1 argument */
```

Comment:
We shouldn't deprecate the current signatures just yet. We need to talk more about a plan to eventually remove the less-strict option if that's the long-term goal. Deprecating early is just going to cause lots of warnings and noise.

Reply:
Instead of deprecation, would it be more desirable to make strictness opt-in first? https://github.com/david-shortman/platform/commit/849e569cba8a511bab505f08d2c594ba73c5b36d

Reply:
If it doesn't disrupt existing code, then yes

Reply:
Ok, I'll look to submit with that approach then.
I guess I could also include a schematic to opt-in all existing selectors, too.

Reply:
I don't think we need to try to migrate existing code at this point. In general I'd like the experience for testing selectors to be the same as today, just with the added benefit of type checking the projector

Reply:
I'm now wondering if this is an opportunity to introduce a more clear API for createSelector which comes with the bonus of strict projectors https://github.com/ngrx/platform/discussions/3098#discussioncomment-1626403

Reply:
We're not planning to introduce any new APIs for selectors